logstash-core/spec/logstash/output_delegator_spec.rb

```diff
@@ -80,6 +88,7 @@
           allow(out_inst).to receive(:metric=).with(any_args)
           allow(out_inst).to receive(:id).and_return("a-simple-plugin")
           allow(out_klass).to receive(:workers_not_supported?).and_return(false)
+          subject.register
```

Comment:
not sure I understand the duplication of some of this `allow` and `register` calls here, can you elaborate a bit more?